onnxruntime/core/optimizer/propagate_cast_ops.cc

```diff
@@ -121,18 +118,39 @@ static bool IsRelevantOutput(const Node* node, const NodeArg* output) {
 
   auto it = opcode_to_output_map.find(node->OpType());
   if (it != opcode_to_output_map.cend()) {
-    const std::vector<int>& selected_outputs = it->second;
+    const auto& selected_outputs = it->second;
     int input_index = optimizer_utils::IndexOfNodeOutput(*node, *output);
     return std::find(selected_outputs.begin(), selected_outputs.end(), input_index) != selected_outputs.end();
   }
   return true;
 }
 
+namespace {
+// borrowed from providers/common.h
+template <class AssociativeContainer, class Key>
+inline bool Contains(const AssociativeContainer& container, const Key& key) {
+  return container.find(key) != container.end();
+}
+}  // namespace
+
 // Check whether the given opcode is fp16 allowed for the given level of optimization.
-static bool IsFP16Allow(const std::string& op_type, size_t level, const FP16AllowOps& fp16_allow_ops) {
-  bool fp16_allow = false;
-  for (size_t i = 0; i <= level && i < fp16_allow_ops.size() && !fp16_allow; ++i) {
-    fp16_allow = fp16_allow_ops[i].find(op_type) != fp16_allow_ops[i].end();
+static bool IsFP16Allow(const std::string& op_type, size_t level, const FP16AllowOps& fp16_allow_level0_ops) {
+  // XXX: Shall we add a check for unsupported level or just ignore it as the current code does?
+  constexpr size_t MaxSupportedCastPropagationLevel = 2;
+
+  using OpsSetType = InlinedHashSet<std::string_view>;
+  static const OpsSetType level1_fp16_allow_set =
+      {"Expand", "Transpose", "Relu", "Reshape", "Split", "Tanh", "Squeeze", "Unsqueeze"};
```

Comment:
These sets used to be constexp (compile time constants). Why are they changed to static const? Too many changes are going into this one PR. It should be spread out over multiple PRs. Majority of the changes are about inlined containers. Other changes should go into a separate PR.

Reply:
`InlinedHashSet` does not have `constexpr` constructors and it is obvious why. However, it is not important any longer because these statics are no longer global.